src/DssSpell.t.base.sol

```diff
@@ -266,22 +266,53 @@ contract DssSpellTestBase is Config, DssTest {
     function setUp() public {
         setValues(address(chief));
 
-        spellValues.deployed_spell_created = spellValues.deployed_spell != address(0) ? spellValues.deployed_spell_created : block.timestamp;
+        if (spellValues.deployed_spell != address(0)) {
+            spell = DssSpell(spellValues.deployed_spell);
+            if (spell.eta() != 0) {
+                // if we have a scheduled spell in the config
+                // we want to roll our fork to the block where it was deployed
+                // this means the test suite will continue to accurately pass/fail
+                // even if mainnet has already scheduled/cast the spell
+                vm.makePersistent(address(this));
+                vm.makePersistent(address(rates));
+                vm.makePersistent(address(addr));
+                vm.makePersistent(address(deployers));
+                _rollToDeployedSpellBlock(address(spell));
+            }
+        } else {
+            spell = new DssSpell();
+        }
+
         _castPreviousSpell();
-        spell = spellValues.deployed_spell != address(0) ?
-            DssSpell(spellValues.deployed_spell) : new DssSpell();
-
-        if (spellValues.deployed_spell_block != 0 && spell.eta() != 0) {
-            // if we have a deployed spell in the config
-            // we want to roll our fork to the block where it was deployed
-            // this means the test suite will continue to accurately pass/fail
-            // even if mainnet has already scheduled/cast the spell
-            vm.makePersistent(address(this));
-            vm.makePersistent(address(rates));
-            vm.makePersistent(address(addr));
-            vm.makePersistent(address(deployers));
-            vm.rollFork(spellValues.deployed_spell_block);
+    }
+
+    function _rollToDeployedSpellBlock(address _spell) internal {
+        if (_spell == address(0)) {
+            vm.rollFork(block.number);
+            return;
+        }
+
+        uint256 low = 5273074; // MCD_VAT Deployed Aug-06-2021 04:26:38 PM +UTC
+        uint256 high = block.number;
+        uint256 mid;
+
+        while (low < high) {
+            mid = (low & high) + (low ^ high) / 2; // rounds down
```

Comment:
Why we say that the simple `(low + high) / 2;` doesn't round down?

Reply:
yes, it rounds down to the closest integer, the other way it rounds down towards zero, truncate (I should have specified that).

Reply:
Didn't understand what you mean. Rounds towards zero?

Reply:
Maybe with some concrete examples for both ways I can understand what you mean.

Reply:
```suggestion
            mid = (low & high) + (low ^ high) / 2; // rounds down to zero, truncate
```

Reply:
???

Reply:
see [here](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/v4.6.0/contracts/utils/math/Math.sol#L29)

Reply:
Yeah ok, that's a complete different explanation from what you mentioned above.
It doesn't have anything to do with one "rounding down to the closest integer" vs the other "rounding down towards zero, truncate" (as they both mean the same exact same thing).
They both do the exact same thing, rounds down to the closest integer or rounds down towards zero.
The only difference is one avoids a possible overflow. So let's try to understand the reasons before doing a change and explain it.

In this specific case, the overflow won't happen in practice, so maybe it is just better to keep the simpler form. To be honest I would need to start digging in order to understand why this math does the same thing (just glancing I don't get it). Maybe as it is part of a known library we can just leave it.

Reply:
testing on remix I got both implementation rounding towards zero or truncate and checking [solc docs](https://docs.soliditylang.org/en/v0.8.16/types.html#division) that's seems the case, maybe the issue I was having here was overflow.

Reply:
how do you have an overflow summing two block numbers?

Reply:
so, testing again against the blocks above, where I noticed the strange pattern (pass, fail, stuck) it seems to pass now with both implementation, so the issue was probably the comparison and the mid bump off by one, but anyway, I think the proposed implementation looks better for general purposes, in case the binary search pattern will be reused.